File: problems/088-merge-sorted-array/solution.c
void merge(
    int* nums1, 
    int nums1Size, 
    int m, 
    int* nums2, 
    int nums2Size, 
    int n) 
{   
    int p1 = m - 1;
    int p2 = n - 1;

    for (int i = m + n - 1; i >= 0; i--) {
        if (p1 >= 0 && (p2 < 0 || nums1[p1] >= nums2[p2])) {
            nums1[i] = nums1[p1];
            p1--;
        } else {
            nums1[i] = nums2[p2];
            p2--;
        }
    }
}

File: problems/088-merge-sorted-array/solution.py
class Solution:
    def merge(self, nums1: list[int], m: int, nums2: list[int], n: int) -> None:
        p1 = m - 1
        p2 = n - 1
        for i in range(m + n - 1, -1, -1):
            if p1 >= 0 and (p2 < 0 or nums1[p1] >= nums2[p2]):
                nums1[i] = nums1[p1]
                p1 -= 1
            else:
                nums1[i] = nums2[p2]
                p2 -= 1